backend/controllers/artworks.ts

```diff
@@ -1,91 +1,115 @@
 import { NextFunction, Request, Response } from "express"
 import mongoose from "mongoose"
-
 const Artwork = require("../models/artwork")
-const Category = require("../models/category")
-const jwt = require("jsonwebtoken")
-
-const getArtwork = async (req: Request, res: Response, next: NextFunction) => {
-    const artworkId = req.params.artworkId
-    try {
-        if (!mongoose.isValidObjectId(artworkId)) {
-            return res.status(400).json(`Invalid artwork id: ${artworkId}`)
-        }
 
-        const artwork = await Artwork.findById(artworkId).exec()
+import { authAsyncWrapper } from "../middleware/auth"
 
-        if (!artwork) {
-            return res.status(404).json("Artwork not found")
-        } else {
-            const columnNames = Category.find({ collectionId: artwork.collectionId }).exec()
-            return res.status(200).json({ artwork, columnNames })
+export const getArtwork = async (req: Request, res: Response, next: NextFunction) => {
+    const artworkId = req.params.artworkId
+    if (!mongoose.isValidObjectId(artworkId)) {
+        const err = new Error(`Invalid artwork id: ${artworkId}`)
+        res.status(400)
+        return next(err)
+    } else {
+        try {
+            const artwork = await Artwork.findById(artworkId).exec()
+            if (!artwork) {
+                const err = new Error(`Artwork ${artworkId} not found`)
+                res.status(404)
+                return next(err)
+            } else {
+                return res.status(200).json({ artwork })
+            }
```

Comment:
Takie podejście może skrócić w wielu miejscach kod ;) Sprawdziłbym tylko czy to tak zadziała. 
```suggestion
            const artwork = await Artwork.findById(artworkId).exec()
            if (!artwork) 
                return res.status(404).json({ error: `Artwork not found` })
            return res.status(200).json({ artwork })
```